### ISSUE TITLE
Higher level placement specification aka label selector based placement

### ISSUE THREAD

**Issue by irfanurrehman** [kubernetes-sigs/federation-v2#87]
Per resource placement, where the corresponding placement resource specifies a list of clusters to place the template is already available. It also makes a lot of sense to have a default behavior for all resources in the abscence of this per object placement specification (as alluded in #75).
   There additionally is a need for a higher level placement intent that a user can specify. Although there could be more extensive placement policies that can be defined, this issue lists the need of a simple label selector based placement intent useful to the end user.

A simple spec for this resource (names could change) could be as below:
```
apiVersion: placement.federation.k8s.io/v1alpha1
kind: Preference
metadata:
  namespace: my-ns
  name: my-placement-preference
spec:
  # Select resources in the namespace whose labels match
  resourceSelector: app=myapp
  # Select labelled clusters
  clusterSelector: region=eu, zone=east
```

**Comment by marun** [comment]
This is definitely a requested feature.  I'm interested in seeing discussion of potential strategies of how to resolve the potential for conflict that selector-based placement could represent.  For example, if a policy-imposed placement preference wants to restrict a resource to clusters in the eu region, and another selector-based placement wants to put that resource in all regions, how would this conflict be detected and resolved automatically or surfaced to the user for manual resolution?

**Comment by irfanurrehman** [comment]
> This is definitely a requested feature. I'm interested in seeing discussion of potential strategies of how to resolve the potential for conflict that selector-based placement could represent.

This is something like a feature ask for Huawei customers; I do not however have a clear solution yet. The intention of creating this issue actually is to trigger discussions on a possible solution.

**Comment by marun** [comment]
This issue was superseded and resolved by #401.
